containers/Dockerfile.buildenv

```diff
@@ -219,10 +219,10 @@ RUN git clone --single-branch --branch v6.10.2 --depth 1 \
         https://github.com/UIUC-PPL/charm charm_6_10_2 \
     && cd /work/charm_6_10_2 \
     && git checkout v6.10.2 \
-    && ./build charm++ multicore-linux-x86_64 gcc ${PARALLEL_MAKE_ARG} -g -O0 \
-    && ./build LIBS multicore-linux-x86_64 gcc ${PARALLEL_MAKE_ARG} -g -O0 \
-    && ./build charm++ multicore-linux-x86_64 clang ${PARALLEL_MAKE_ARG} -g -O0 \
-    && ./build LIBS multicore-linux-x86_64 clang ${PARALLEL_MAKE_ARG} -g -O0 \
+    && ./build LIBS multicore-linux-x86_64 gcc \
```

Comment:
why build the shared libs here? Linux allows you to mix shared and static libs as long as the static libs are built with `-fPIC` and our build system doesn't look for Charm++'s shared libs (because they stick them in the wrong directory)

Reply:
The Charm docs say to compile shared libs: https://github.com/UIUC-PPL/charm#building-dynamic-libraries. Then we can pass `-charm-shared` to `charmc --print-building-blocks` and get the compiler configuration for shared libs. If we don't do both (have the Charm shared libs compiled and pass the `-charm-shared` flag) then I don't think they guarantee that things will work.

Reply:
Sounds good!